docs/tutorials/datastream_api.md

```diff
@@ -0,0 +1,252 @@
+            this.name = name;
+            this.age = age;
+        };
+
+        public String toString() {
+            return this.name.toString() + ": age " + this.age.toString();
+        };
+    }
+}
+{% endhighlight %}
+
+### Stream execution environment
+
+Every Flink application needs an execution environment, `env` in this example. Streaming
+applications should use a `StreamExecutionEnvironment`.
+
+The DataStream API calls made in your application build a job graph that is attached to the
+`StreamExecutionEnvironment`. When `env.execute()` is called this graph is packaged up and sent to
+the Flink Master, which parallelizes the job and distributes slices of it to the Task Managers for
+execution. Each parallel slice of your job will be executed in a *task slot*.
+
+Note that if you don't call execute(), your application won't be run.
+
+<img src="{{ site.baseurl }}/fig/distributed-runtime.svg" alt="Flink runtime: client, job manager, task managers" class="offset" width="80%" />
+
+This distributed runtime depends on your application being serializable. It also requires that all
+dependencies are available to each node in the cluster.
+
+### Basic stream sources
+
+The example above constructs a `DataStream<Person>` using `env.fromElements(...)`. This is a
+convenient way to throw together a simple stream for use in a prototype or test. There is also a
+`fromCollection(Collection)` method on `StreamExecutionEnvironment`. So instead, you could do this:
+
+{% highlight java %}
+List<Person> people = new ArrayList<Person>();
+
+people.add(new Person("Fred", 35));
+people.add(new Person("Wilma", 35));
+people.add(new Person("Pebbles", 2));
+
+DataStream<Person> flintstones = env.fromCollection(people);
+{% endhighlight %}
+
+Another convenient way to get some data into a stream while prototyping is to use a socket
+
+{% highlight java %}
+DataStream<String> lines = env.socketTextStream("localhost", 9999)
+{% endhighlight %}
+
+or a file
+
+{% highlight java %}
+DataStream<String> lines = env.readTextFile("file:///path");
+{% endhighlight %}
+
+In real applications the most commonly used data sources are those that support low-latency, high
+throughput parallel reads in combination with rewind and replay -- the prerequisites for high
+performance and fault tolerance -- such as Apache Kafka, Kinesis, and various filesystems. REST APIs
+and databases are also frequently used for stream enrichment.
+
+### Basic stream sinks
+
+The example above uses `adults.print()` to print its results to the task manager logs (which will
+appear in your IDE's console, when running in an IDE). This will call `toString()` on each element
+of the stream.
+
+The output looks something like this
+
+    1> Fred: age 35
+    2> Wilma: age 35
+
+where 1> and 2> indicate which sub-task (i.e., thread) produced the output.
+
+You can also write to a text file
+
+{% highlight java %}
+stream.writeAsText("/path/to/file")
```

Comment:
seems to be deprecated now; not sure a full `StreamingFileSink` (as proposed) would be a good replacement for the training though